src/OnChain.ts

```diff
@@ -189,10 +187,17 @@ export const OnChainMixin = (superclass) =>
             }
 
             await lockExtendOrThrow({ lock, logger: onchainLoggerOnUs }, async () => {
-              return await MainBook.entry()
-                .credit(accountPath(payeeUser._id), sats, metadata)
-                .debit(this.user.accountPath, sats, { ...metadata, memo })
-                .commit()
+              const tx = await ledger.addOnUsPayment({
+                description: "",
+                sats,
+                metadata,
+                payerUser: this.user,
+                payeeUser,
+                memoPayer: memo,
+                shareMemoWithPayee: false,
+                lastPrice: UserWallet.lastPrice,
```

Comment:
this is something we will eventually want to refactor (in later PR) not sure the price should be attached to the wallet any more the way it currently is

Reply:
yes, maybe the signatures of that methods will change when we add usd balance/invoices